Metamist ETL integration example.

milo-hyben opened the pull request.

milo-hyben commented on metamist_infrastructure/driver.py:
```diff
@@ -550,7 +550,15 @@ def _setup_etl_pubsub(self):
     def etl_extract_function(self):
         """etl_extract_function"""
-        return self._etl_function('extract', self.etl_extract_service_account)
+        custom_audience_list = [
```
We should make this as config for metamist-etl-extract function.

illusional commented:
There's one here, do you mind moving or updating this example?

https://github.com/populationgenomics/metamist/blob/dev/scripts/etl_caller_example.py

milo-hyben commented:
OK, let me update that one in the script folder :-)

milo-hyben commented:
Hey @illusional, I have updated both script and as well converted the custom audience list as a config, added to both cog-infra/cpg-infra-private:

https://github.com/populationgenomics/cpg-infrastructure/pull/209
https://github.com/populationgenomics/cpg-infrastructure-private/pull/292

illusional commented on scripts/etl_caller_example.py:
```diff
@@ -14,8 +15,8 @@
-URL = 'https://metamist-etl-mnrpw3mdza-ts.a.run.app'
-TYPE = 'NAME_OF_EXTERNAL_PARTY/v1'
+URL = 'https://metamist-extract.popgen.rocks'
+TYPE = 'bbv/v1'  # 'NAME_OF_EXTERNAL_PARTY/v1'
```
```suggestion
TYPE = 'NAME_OF_EXTERNAL_PARTY/v1'
```

illusional commented on metamist_infrastructure/driver.py:
```diff
@@ -646,6 +649,13 @@ def _etl_function(
 
+        # prepare custom audience_list
+        custom_audience_list = None
+        if self.config.metamist.etl.custom_audience_list:
+            custom_audience_list = json.dumps(
+                self.config.metamist.etl.custom_audience_list.get(f_name)
+            )
```
```suggestion
        if self.config.metamist.etl.custom_audience_list and self.config.metamist.etl.custom_audience_list.get(f_name):
            custom_audience_list = json.dumps(
                self.config.metamist.etl.custom_audience_list.get(f_name)
            )
```

illusional commented:
Noting you'll need to merge this, AND update the cpg-infra requirements.txt pointer to this repo
